at.medevit.ch.artikelstamm.elexis.common/src/at/medevit/ch/artikelstamm/elexis/common/ui/cv/FilterBlackboxAction.java

```diff
@@ -0,0 +1,57 @@
+package at.medevit.ch.artikelstamm.elexis.common.ui.cv;
+
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.wb.swt.ResourceManager;
+
+import at.medevit.ch.artikelstamm.BlackBoxReason;
+import ch.artikelstamm.elexis.common.ArtikelstammItem;
+import ch.elexis.core.ui.actions.PersistentObjectLoader.QueryFilter;
+import ch.elexis.core.ui.icons.Images;
+import ch.elexis.data.PersistentObject;
+import ch.elexis.data.Query;
+
+/**
+ * Filter blackboxed items out. 
+ * @see BlackBoxReason
+ * @author gerry
+ *
+ */
+public class FilterBlackboxAction extends Action {
+	ArtikelstammFlatDataLoader fdl;
+	BlackBoxQueryFilter bbqf;
+	private ImageDescriptor blackBoxedImage = ResourceManager.getPluginImageDescriptor("at.medevit.ch.artikelstamm.ui",
+			"/rsc/icons/flag-black.png");
+	
+	public FilterBlackboxAction(ArtikelstammFlatDataLoader fdl){
+		super("Backbox filtern",Action.AS_CHECK_BOX);
```

Comment:
BTW - thats a nice one :) "Backbox"

Reply:
Yes, that's the one big difference to the medelexis implementation :-)